### PR TITLE
Updating description for MatchMode in Migration Rules

### PR DESCRIPTION
**Pull request by lapel22**
For https://jira.devtools.intel.com/browse/DOC-12840

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: docs/dev_guide/migration/migration-rules.rst [hunk 1]
```diff
@@ -62,8 +62,8 @@ Migration rules are specified in YAML files. A single rule file may contain mult
      - String value
      - Required. Specify the CMake syntax name that will be migrated. Use the unique name for the CMake syntax.
    * - MatchMode
-     - ``Partial`` | ``Full``
-     - Required. Specify the match mode with full word match or partial word match. If not specified, partial match mode will be used.
+     - ``Partial`` | ``Full`` | ``StrictFull``
+     - Optional. Specify the match mode with full word match, strict full word match, or partial word match. If not specified, partial match mode will be used. The partial matching mode means the matching string can be surrounded by arbitrary characters including whitespace. The full matching mode means the matching string can only be surrounded by non-identifier characters (letters, numbers and underscore). The strict matching mode means the matching string can only be surrounded by whitespace characters.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/migration-rules.rst]: @tomflinda ,  please update the the doc.

Comment (tomflinda) [inline docs/dev_guide/migration/migration-rules.rst]: Pls update the sub string:
`The partial matching mode means the matching string can be surrounded by arbitrary characters including whitespace. The full matching mode means the matching string can only be surrounded by non-identifier characters (letters, numbers and underscore). The strict matching mode means the matching string can only be surrounded by whitespace characters.`
to
`The partial matching mode means the matched string can be surrounded by arbitrary characters including whitespace.
The full matching mode means the matched string should not be surrounded by identifier characters (letters, numbers, and underscore).
The strict matching mode means the matched string should only be surrounded by whitespace characters.`

Comment (tomflinda) [reply docs/dev_guide/migration/migration-rules.rst]: Okay

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**